src/polycap-source.c

```diff
@@ -633,6 +633,26 @@ polycap_transmission_efficiencies* polycap_source_get_transmission_efficiencies(
 		#pragma omp critical
 		{
 		sum_irefl += photon->i_refl;
+		//Assign memory to arrays holding leak photon information (and fill them)
+		//TODO: If we can find a way to do this outside of critical block... would be opportune for calculation speeds
+		sum_nleaks += photon->n_leaks;
+		if(sum_nleaks > 0 && photon->n_leaks > 0){
+			efficiencies->images->leak_coords[0] = realloc(efficiencies->images->leak_coords[0], sizeof(double)* sum_nleaks);
+			efficiencies->images->leak_coords[1] = realloc(efficiencies->images->leak_coords[1], sizeof(double)* sum_nleaks);
+			efficiencies->images->leak_coords[2] = realloc(efficiencies->images->leak_coords[2], sizeof(double)* sum_nleaks);
+			efficiencies->images->leak_dir[0] = realloc(efficiencies->images->leak_dir[0], sizeof(double)* sum_nleaks);
+			efficiencies->images->leak_dir[1] = realloc(efficiencies->images->leak_dir[1], sizeof(double)* sum_nleaks);
+			efficiencies->images->leak_coord_weights = realloc(efficiencies->images->leak_coord_weights, sizeof(double)*n_energies* sum_nleaks);
```

Comment:
Bij deze `realloc` en bovenstaande geeft valgrind errors over `definately lost bytes`. Kzie echter niet direct wat ik fout doe..
Mocht je overigens een manier weten hoe ik deze informatie elegant kan opslaan buiten de `critical block` is het ook welkom. Probleem is voornamelijk dat ik op voorhand geen idee heb hoeveel 'leak photons' er zullen zijn. (dit kunnen er in principe zelfs meer zijn dan het maximaal aantal gesimuleerde fotonen, daar elk gesimuleerd foton theoretisch gezien op elke reflectie plaats zou kunnen lekken. Beeld je een monocapillair in een foton van 50 keV dat doorheen het capillair botst, maar dus ook op elke reflectie een portie doorheen het capillair lekt)

Reply:
Hm... net idee gehad... Ik free die arrays (nog) niet in `polycap_transmission_efficiencies_free()`, mss heeft hij daar een probleem mee. Will test.

Reply:
Hey Pieter,


Als je het aantal `realloc`'s daar wil verminderen raad ik u aan om te werken met een array lengte en een array capaciteit. De capaciteit kan je dan pas gaan verhogen met een realloc wanneer de lengte groter is geworden dan de capaciteit. Maar ipv de capaciteit altijd met 1 te verhogen, kies je best voor een verdubbeling van de huidige capaciteit. Dit zou kunnen een redelijk groot verschil maken voor performantie.

Ik doe iets gelijkaardig hier: 
https://github.com/PieterTack/polycap/blob/7fac63874c7b0a2d3e7d087d2710904bb485544b/src/polycap-transmission-efficiencies.c#L50-L59

Reply:
Ahja, eenzelfde systeem gebruiken ze inderdaad in dat programmaatje dat ik ook gebruik waarin ze strings van ongekende lengte inlezen :) Good thinking! Zal het zodra eens proberen implementeren, als ik er de tijd nog voor vind. Morgen en volgende week verlof, dus zal dan iets minder actief zijn :)

Reply:
Goeie Gentse Feesten!